src/main/java/picard/illumina/ClusterDataToSamConverter.java

```diff
@@ -178,10 +179,10 @@ private SAMRecord createSamRecord(final ReadData readData, final String readName
 
         if (!molecularIndexes.isEmpty()) {
             if (!this.molecularIndexTag.isEmpty()) {
-                sam.setAttribute(this.molecularIndexTag, String.join("", molecularIndexes));
+                sam.setAttribute(this.molecularIndexTag, String.join(molecularIndexDelimiter, molecularIndexes));
             }
             if (!this.molecularIndexQualityTag.isEmpty()) {
-                sam.setAttribute(this.molecularIndexQualityTag, String.join("", molecularIndexQualities));
+                sam.setAttribute(this.molecularIndexQualityTag, String.join(molecularIndexDelimiter, molecularIndexQualities));
```

Comment:
given that `-` is a legal quality character, splitting the quality string will involve splitting the bases first and then using the resulting lengths to split the quality string, I wonder if it would be clearer without the delimiter in the quality string. Not a request, just a opening for discussion.